src/core/QCheck.ml

```diff
@@ -763,16 +779,19 @@ module Shrink = struct
     | None -> ()
     | Some shrink -> list_elems shrink l yield
 
-  let string s yield =
+  let _shrink_string_and_bytes fold get s yield =
     let buf = Buffer.create 42 in
     list ~shrink:char
-      (string_fold_right (fun c acc -> c::acc) s [])
+      (fold (fun c acc -> c::acc) s [])
       (fun cs ->
          List.iter (fun c -> Buffer.add_char buf c) cs;
-         let s = Buffer.contents buf in
+         let s = get buf in
          Buffer.clear buf;
          yield s)
 
+  let bytes = _shrink_string_and_bytes bytes_fold_right (fun b -> Buffer.contents b |> Bytes.of_string)
```

Comment:
There's `Buffer.to_bytes` since 4.02 which I think can simplify the above to:
```ocaml
let bytes = _shrink_string_and_bytes bytes_fold_right Bytes.to_bytes
```